### PR TITLE
Update Developer Guide

### LINKED ISSUE #125: Update Developer Guide [Kyriel]
The relevant sections are: 

- Delete Command
- Miscellaneous documentation

Please ensure that the final state of the DG reflects Flagship properly.

### PR DESCRIPTION
**Pull request by s-kybound**
Resolves #125

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
 
-[![CI Status](https://github.com/se-edu/addressbook-level3/workflows/Java%20CI/badge.svg)](https://github.com/se-edu/addressbook-level3/actions)
-[![codecov](https://codecov.io/gh/se-edu/addressbook-level3/branch/master/graph/badge.svg)](https://codecov.io/gh/se-edu/addressbook-level3)
+[![CI Status](https://github.com/AY2324S1-CS2103T-W17-1/tp/workflows/Java%20CI/badge.svg)](https://github.com/AY2324S1-CS2103T-W17-1/tp/actions)
+[![codecov](https://codecov.io/gh/AY2324S1-CS2103T-W17-1/tp/branch/master/graph/badge.svg)](https://codecov.io/gh/AY2324S1-CS2103T-W17-1/tp)
 
 ![Ui](images/Ui.png)
 
-**AddressBook is a desktop application for managing your contact details.** While it has a GUI, most of the user interactions happen using a CLI (Command Line Interface).
+**Flagship is a desktop application used to help aspiring students track internship applications.** he user interacts with it using a CLI, and it has a GUI created with JavaFX. It is written in Java.
```
Comment (jinyang628) [inline docs/index.md:14]: I think you meant 'the' user

**Review by jinyang628 — CHANGES_REQUESTED** [review]
Minor typo - but otherwise good work kyriel. Ping me when you have rectified the typo

File: docs/index.md (hunk 1 above)
Comment (s-kybound) [reply docs/index.md:14]: Resolved.

**Review by jinyang628 — APPROVED** [review]
LGTM. Thanks Kyriel

**Pull request merged**